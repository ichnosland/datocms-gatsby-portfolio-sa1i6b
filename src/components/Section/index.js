/**
*
* Section
*
*/

import styled, { css } from 'styled-components';

import media from 'style/mediainjector';
import nuvole from './images/nuvole.png';

export const Section = styled.section`
  background-color: ${(props) => props.bgColor ? props.bgColor : 'transparent'};
  width: 100%;
  height: ${(props) => props.height ? props.height : 'auto'};
  display: flex;
  justify-content: ${(props) => props.justify ? props.justify : 'center'};
  align-items: center;
  flex-direction: ${(props) => props.direction ? props.direction : 'row'};
  position: relative;
  padding: ${(props) => props.padding ? props.padding : '50px'};
  ${(props) => {
    const padding = `padding: ${props.paddingmobile || '30px 20px'};`;
    return media.lt480([padding]);
  }}
  ${({ full }) => full && css`
    height: 100%;
    flex-direction: ${(props) => props.direction ? props.direction : 'column'};
    ${(props) => props.flextop && css`
      justify-content: space-between;
      &:before {
        content: '';
        display: block;
      }
    `};
    ${(props) => props.flexbottom && css`
      justify-content: space-between;
      &:after {
        content: '';
        display: block;
      }
    `};
  `};
  ${(props) => props.esecuzione && media.mobile`
    justify-content: flex-start;
    padding: 0;
    `};
  ${(props) => props.first && css`
    padding-top: 156px;
    ${media.lt667`
      padding-top: 76px;
      `}
    `};
  `;

export const SkySection = styled.section`
  position: relative;
  min-height: 530px;
  padding-top: 126px;
  margin-bottom: -5px;
  background-image: url(${nuvole}), linear-gradient(0deg, rgba(255,255,255,0.00) 33%, rgba(0,171,229,0.65) 100%);
  background-size: 1078px 139px, 100%;
  background-repeat:  no-repeat, no-repeat;
  background-position: center 30%, center;
  ${media.lt667`
    padding-top: 76px;
    background-image: url(${nuvole}), linear-gradient(0deg, rgba(255,255,255,0.00) 33%, rgba(0,171,229,0.15) 100%);
  `}
  `;

export const FirstSection = styled.section`
  position: relative;
  min-height: ${(props) => props.minHeight};
  padding-top: 126px;
  ${media.lt667`
    padding-top: 76px;
  `}
`;

export default Section;
