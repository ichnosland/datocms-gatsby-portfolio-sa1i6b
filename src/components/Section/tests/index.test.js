import React from 'react';
import { shallow } from 'enzyme';

import Section, { SkySection, FirstSection } from '../index';

describe('<Section />', () => {
  it('should render a <section> tag', () => {
    const renderedComponent = shallow(<Section />);
    expect(renderedComponent.type()).toEqual('section');
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<Section id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should render its text', () => {
    const children = 'Text';
    const renderedComponent = shallow(
      <Section>{children}</Section>
    );
    expect(renderedComponent.contains(children)).toBe(true);
  });

  it('should render its css', () => {
    const renderedComponent = shallow(<Section />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props bgColor', () => {
    const bgColor = 'blue';
    const renderedComponent = shallow(<Section bgColor={bgColor} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props height', () => {
    const height = '30px';
    const renderedComponent = shallow(<Section height={height} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props justify', () => {
    const justify = 'space-between';
    const renderedComponent = shallow(<Section justify={justify} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props direction', () => {
    const direction = 'column';
    const renderedComponent = shallow(<Section direction={direction} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props padding', () => {
    const padding = '30px';
    const renderedComponent = shallow(<Section padding={padding} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props padding mobile', () => {
    const padding = '40px';
    const renderedComponent = shallow(<Section paddingMobile={padding} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props full', () => {
    const renderedComponent = shallow(<Section full />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props full + direction', () => {
    const direction = 'row';
    const renderedComponent = shallow(<Section full direction={direction} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props full + flextop', () => {
    const renderedComponent = shallow(<Section full flextop />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props full + flexbottom', () => {
    const renderedComponent = shallow(<Section full flexbottom />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props esecuzione', () => {
    const renderedComponent = shallow(<Section esecuzione />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props first', () => {
    const renderedComponent = shallow(<Section first />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<SkySection />', () => {
  it('should render a <section> tag', () => {
    const renderedComponent = shallow(<SkySection />);
    expect(renderedComponent.type()).toEqual('section');
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<SkySection id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should render its text', () => {
    const children = 'Text';
    const renderedComponent = shallow(
      <SkySection>{children}</SkySection>
    );
    expect(renderedComponent.contains(children)).toBe(true);
  });

  it('should render its css', () => {
    const renderedComponent = shallow(<SkySection />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<FirstSection />', () => {
  it('should render a <section> tag', () => {
    const renderedComponent = shallow(<FirstSection />);
    expect(renderedComponent.type()).toEqual('section');
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<FirstSection id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should render its css w/ props minHeight', () => {
    const minHeight = '200px';
    const renderedComponent = shallow(<FirstSection minHeight={minHeight} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
