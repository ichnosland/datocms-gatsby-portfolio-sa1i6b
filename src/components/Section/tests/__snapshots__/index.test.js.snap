// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstSection /> should render its css w/ props minHeight 1`] = `
.c0 {
  position: relative;
  min-height: 200px;
  padding-top: 126px;
}

@media (max-width:41.6875em) {
  .c0 {
    padding-top: 76px;
  }
}

<section
  className="c0"
/>
`;

exports[`<Section /> should render its css 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
/>
`;

exports[`<Section /> should render its css w/ props bgColor 1`] = `
.c0 {
  background-color: blue;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
/>
`;

exports[`<Section /> should render its css w/ props direction 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding: 50px;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
  direction="column"
/>
`;

exports[`<Section /> should render its css w/ props esecuzione 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

@media (max-width:64em) {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding: 0;
  }
}

<section
  className="c0"
/>
`;

exports[`<Section /> should render its css w/ props first 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
  padding-top: 156px;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

@media (max-width:41.6875em) {
  .c0 {
    padding-top: 76px;
  }
}

<section
  className="c0"
/>
`;

exports[`<Section /> should render its css w/ props full + direction 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
  height: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
  direction="row"
/>
`;

exports[`<Section /> should render its css w/ props full + flexbottom 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0:after {
  content: '';
  display: block;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
/>
`;

exports[`<Section /> should render its css w/ props full + flextop 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0:before {
  content: '';
  display: block;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
/>
`;

exports[`<Section /> should render its css w/ props full 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
/>
`;

exports[`<Section /> should render its css w/ props height 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
  height="30px"
/>
`;

exports[`<Section /> should render its css w/ props justify 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
/>
`;

exports[`<Section /> should render its css w/ props padding 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 30px;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
/>
`;

exports[`<Section /> should render its css w/ props padding mobile 1`] = `
.c0 {
  background-color: transparent;
  width: 100%;
  height: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  padding: 50px;
}

@media (max-width:30em) {
  .c0 {
    padding: 30px 20px;
  }
}

<section
  className="c0"
/>
`;

exports[`<SkySection /> should render its css 1`] = `
.c0 {
  position: relative;
  min-height: 530px;
  padding-top: 126px;
  margin-bottom: -5px;
  background-image: url(IMAGE_MOCK),linear-gradient(0deg,rgba(255,255,255,0.00) 33%,rgba(0,171,229,0.65) 100%);
  background-size: 1078px 139px,100%;
  background-repeat: no-repeat,no-repeat;
  background-position: center 30%,center;
}

@media (max-width:41.6875em) {
  .c0 {
    padding-top: 76px;
    background-image: url(IMAGE_MOCK),linear-gradient(0deg,rgba(255,255,255,0.00) 33%,rgba(0,171,229,0.15) 100%);
  }
}

<section
  className="c0"
/>
`;
